### PR TITLE
Added subscriptionGift to subscribeHandler

### PR DESCRIPTION
**Pull request by CreativeCodeCat**
Added subscriptionGift to subscribeHandler on twitches end will work on discord next :)

### CODE REVIEW

**Comment by BrandenB** [comment]
**Issues:**
- No way to set a reward for the user who gifted the subscription.
- No commands to change these new settings.

**Comment by CreativeCodeCat** [comment]
ok ill get them done then push it to this pull request

**Comment by CreativeCodeCat** [comment]
There commands made and added a setting for the reward for the gifter

**Comment by BrandenB** [comment]
Did you test this? I don't see any code adding the reward to the user who gave the subscription, also the `!giftsubwelcometoggle` command isn't using the right key in the database. Please be sure to test everything. 

Thanks.

**Comment by CreativeCodeCat** [comment]
Fixed sorry i forgot to add that whn i added the customization for it thansk for letting me know also hopw could i test this as im not an affilate or partner

File: javascript-source/handlers/subscribeHandler.js [hunk 1]
```diff
@@ -233,6 +311,20 @@
...
+        /**
+         * @commandpath giftsubreward [points] - Set an award for resubscribers.
+         */
+        if (command.equalsIgnoreCase('giftsubreward')) {
+            if (isNaN(parseInt(action))) {
+                $.say($.whisperPrefix(sender) + $.lang.get('subscribehandler.giftsub.reward.usage'));
+                return;
+            }
+            
+            reSubReward = parseInt(action);
```
Comment (BrandenB) [inline javascript-source/handlers/subscribeHandler.js]: This should be `giftSubReward` and not `reSubReward `.

File: javascript-source/handlers/subscribeHandler.js [hunk 2]
```diff
@@ -204,6 +268,20 @@
...
+        /*
+         * @commandpath giftsubmessage [message] - Set a message for resubscribers.
+         */
+        if (command.equalsIgnoreCase('giftsubmessage')) {
+            if (action === undefined) {
+                $.say($.whisperPrefix(sender) + $.lang.get('subscribehandler.giftsub.msg.usage'));
+                return;
+            }
+
+            giftSubMessage = argsString ;
```
Comment (BrandenB) [inline javascript-source/handlers/subscribeHandler.js]: Remove the extra space after `argsString`.

File: javascript-source/handlers/subscribeHandler.js [hunk 3]
```diff
@@ -162,6 +217,15 @@
             $.say($.whisperPrefix(sender) + (reSubWelcomeToggle ? $.lang.get('subscribehandler.resub.toggle.on') : $.lang.get('subscribehandler.resub.toggle.off')))
         }
 
+        /*
+         * @commandpath giftsubwelcometoggle - Enable or disable subgifting alerts.
+         */
+        if (command.equalsIgnoreCase('giftsubwelcometoggle')) {
+            giftSubWelcomeToggle = !giftSubWelcomeToggle;
+            $.setIniDbBoolean('subscribeHandler', 'reSubscriberWelcomeToggle', giftSubWelcomeToggle);
```
Comment (BrandenB) [inline javascript-source/handlers/subscribeHandler.js]: The key should be `giftSubWelcomeToggle` and not `reSubscriberWelcomeToggle`.

**Review by BrandenB — CHANGES_REQUESTED** [review]
Please test everything like I mentioned before. This is clearly not fully tested. Just run commands, or send fake events from the core if you want to test everything, or read your code over and over again.

**Comment by CreativeCodeCat** [comment]
How do i send fake events ? im new to all that stuff but i fixed them 3 things you mentioned

**Comment by BrandenB** [comment]
Simple usage of Rhino can accomplish this from scripts if you don't want to mess with the core.

`Packages.tv.phantombot.event.EventBus.instance().post(EVENT_OBJECT_HERE)`

**Comment by CreativeCodeCat** [comment]
If you like you can close this and ill just wait until you guys do it then update :) i just thought id "try" to help out a bit but failed

**Review by BrandenB — APPROVED** [review]

**Pull request merged**